### PR TITLE
wxPython30: fix on darwin

### PR DESCRIPTION
**Pull request by veprbl**
###### Things done

- [x] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [x] macOS
   - [ ] Linux
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by FRidh** [comment]
I think we should send this to staging.

**Comment by veprbl** [comment]
This has only 5 reverse dependencies.

File: pkgs/development/python-modules/wxPython/3.0.nix [hunk 1]
```diff
@@ -27,15 +30,27 @@ buildPythonPackage rec {
...
+    # remove wxPython's darwin hack that interference with python-2.7-distutils-C++.patch
+    substituteInPlace config.py \
+      --replace "distutils.unixccompiler.UnixCCompiler = MyUnixCCompiler" ""
+    # this check is supposed to only return false on older systems running non-framework python
+    substituteInPlace src/osx_cocoa/_core_wrap.cpp \
+      --replace "return wxPyTestDisplayAvailable();" "return true;"
+  '';
 
-  NIX_LDFLAGS = "-lX11 -lgdk-x11-2.0";
+  NIX_LDFLAGS = if stdenv.isDarwin then "" else "-lX11 -lgdk-x11-2.0";
```
Comment (LnL7) [inline pkgs/development/python-modules/wxPython/3.0.nix]: `lib.optionalString (!stdenv.isDarwin)` is a bit nicer for this

File: pkgs/development/python-modules/wxPython/3.0.nix [hunk 2]
```diff
@@ -27,15 +30,27 @@ buildPythonPackage rec {
...
 
   buildPhase = "";
 
   installPhase = ''
-    ${python.interpreter} setup.py install WXPORT=gtk2 NO_HEADERS=1 BUILD_GLCANVAS=${if openglSupport then "1" else "0"} UNICODE=1 --prefix=$out
+    ${python.interpreter} setup.py install WXPORT=${if stdenv.isDarwin then "osx_cocoa" else "gtk2"} NO_HEADERS=1 BUILD_GLCANVAS=${if openglSupport then "1" else "0"} UNICODE=1 --prefix=$out
```
Comment (LnL7) [inline pkgs/development/python-modules/wxPython/3.0.nix]: Do those work as environment variables?

Comment (veprbl) [inline pkgs/development/python-modules/wxPython/3.0.nix]: No

Comment (LnL7) [inline pkgs/development/python-modules/wxPython/3.0.nix]: allright

**Review by LnL7 — APPROVED** [review]

**Comment by veprbl** [comment]
Thank you!

**Pull request merged**